### PR TITLE
Fix hook in SearchPage component

### LINKED ISSUE #11428: Searching from main search page prepends unrelated string to query
#### Steps to reproduce:

1. visit sourcegraph.com/search
2. type any query (eg. `repo:github.com/sourcegraph/sourcegraph`) in the search field
3. submit query

#### Expected behavior:

Search is performed with the query you typed in.

#### Actual behavior:

`,function () { [native code] }` is prepended to the query

image

### PR DESCRIPTION
**Pull request by lguychard**
Fixes #11428

### CODE REVIEW

**Comment by eseliger** [comment]
Closes https://github.com/sourcegraph/sourcegraph/issues/11406

File: web/src/search/input/SearchPage.tsx [hunk 1]
```diff
@@ -78,7 +78,7 @@ export const SearchPage: React.FunctionComponent<Props> = props => {
     })
 
     /** The query that results from combining all values in the query builder form. */
-    const builderQuery = useState('')
+    const [builderQuery] = useState('')
```
Comment (felixfbecker) [inline web/src/search/input/SearchPage.tsx:81]: 🤦 

Thinking about why this happened, if we ever had to _set_ it, that should have uncovered this, and I assumed all state is set somewhere.
But... if we never set this, why was/is it state at all? Either we could just pass a constant empty string, or I missed a `setState()` of this somewhere that needs to be re-added...

Comment (lguychard) [reply web/src/search/input/SearchPage.tsx:81]: I didn't really ask myself that question as I was focused on topically fixing the issue, but it seems like this was never set in the original component: https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24%400c996ec+f:SearchPage.tsx+builderQuery&patternType=literal

Comment (felixfbecker) [reply web/src/search/input/SearchPage.tsx:81]: There definitely was no `setState()` call [before my PR](https://github.com/sourcegraph/sourcegraph/blob/f425fda6e0ad1ef8ba5fdedbf0e755a8c12feec6/web/src/search/input/SearchPage.tsx), so I think this state variable should just be removed completely and replaced with empty strings everywhere.

But I wonder if that was intentional. @attfarhan can you shine light on this?

One thing I imagine is that the query builder used to be in this component, but now it is on it's own independent page, but when refactoring this was not removed.

Comment (attfarhan) [reply web/src/search/input/SearchPage.tsx]: That appears to be what happened: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/commit/7373e0f3cb596e25c1ca7b7c9b04083cd0acc57f#diff-9554276aabc44631d88947764e014726

**Comment by felixfbecker** [comment]
Side note: This is another bug that would have been caught by e2e tests or  integration tests

**Pull request merged**